File: Example/PDStrategy/PDStrategy/PDStrategy/Views/PDButton.m
//
//  PDButton.m
//  PDStrategy
//
//  Created by Pavel Deminov on 19/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

#import "PDButton.h"

@implementation PDButton

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    [self setup];
    return self;
}

- (void)setup {
    self.translatesAutoresizingMaskIntoConstraints = NO;
    self.highlightedColor = [UIColor grayColor];
    self.normalColor = [UIColor whiteColor];
    
    self.layer.borderColor = [UIColor lightGrayColor].CGColor;
    self.layer.borderWidth = 1 / [UIScreen mainScreen].scale;
    self.layer.cornerRadius = 6 / [UIScreen mainScreen].scale;
    [self setTitleColor:[UIColor grayColor] forState:(UIControlStateNormal)];
    [self setTitleColor:[UIColor whiteColor] forState:(UIControlStateHighlighted)];
    [self updateUI];
}

- (void)updateUI {
    self.backgroundColor = self.highlighted ? self.highlightedColor : self.normalColor;
}

@end

File: Example/PDStrategy/PDStrategy/PDStrategy/Views/PDButton.swift
//
//  PDButton.swift
//  PDStrategy
//
//  Created by Pavel Deminov on 11/11/2017.
//  Copyright © 2017 Pavel Deminov. All rights reserved.
//

import UIKit

class PDButton: UIButton {

    var top: NSLayoutConstraint?
    var leading: NSLayoutConstraint?
    var trailing: NSLayoutConstraint?
    var bottom: NSLayoutConstraint?
    
    var width: NSLayoutConstraint?
    var height: NSLayoutConstraint?
    
    var x: NSLayoutConstraint?
    var y: NSLayoutConstraint?
    
    var highlightedColor = UIColor.gray
    var normalColor = UIColor.white
    
    override var isHighlighted:Bool {
        didSet{
            updateUI()
        }
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    init() {
        super.init(frame: CGRect())
        self.setup()
    }
    
    func setup() {
        translatesAutoresizingMaskIntoConstraints = false
        layer.borderColor = UIColor.lightGray.cgColor
        layer.borderWidth = 1 / UIScreen.main.scale
        layer.cornerRadius = 6 / UIScreen.main.scale
        setTitleColor(UIColor.gray, for: .normal)
        setTitleColor(UIColor.white, for: .highlighted)
        updateUI()
    }
    
    func updateUI() {
        self.backgroundColor = isHighlighted ? highlightedColor : normalColor
        
    }

}
